src/algebra/homology/chain_complex.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.graded_object
+import category_theory.differential_object
+
+/-!
+# Chain complexes
+
+We define a chain complex in `V` as a differential `ℤ`-graded object in `V`.
+
+This is fancy language for the obvious definition,
+and it seems we can use it straightforwardly:
+
+```
+example (C : chain_complex V) : C.X 5 ⟶ C.X 6 := C.d 5
+```
+
+We define the forgetful functor to `ℤ`-graded objects, and show that
+`chain_complex V` is concrete when `V` is, and `V` has coproducts.
+-/
+
+universes v u
+
+open category_theory
+open category_theory.limits
+
+section
+variables (V : Type u) [𝒱 : category.{v} V]
+include 𝒱
+
+variables [has_zero_morphisms.{v} V]
+
+/--
+A chain complex in `V` is "just" a differential `ℤ`-graded object in `V`.
+-/
+-- For now the "shift" is fixed to be the +1 direction,
+-- making this a cochain complex, rather than a chain complex.
+def chain_complex : Type (max v u) :=
+differential_object.{v} (graded_object ℤ V)
+
+-- The chain groups of a chain complex `C` are accessed as `C.X i`,
+-- and the differentials as `C.d i : C.X i ⟶ C.X (i+1)`.
+example (C : chain_complex V) : C.X 5 ⟶ C.X 6 := C.d 5
+
+variables {V}
+/--
+A convenience lemma for morphisms of differential graded objects,
+picking out one component of the commutation relation.
+-/
+-- Could this be a simp lemma? Which way?
+lemma category_theory.differential_object.hom.comm_at {X Y : differential_object.{v} (graded_object ℤ V)}
```

Comment:
Why do you prove this up here instead of waiting until you have the category structure on `chain_complex` so that `X` and `Y` can be ob type `chain_complex V`? Why does the third example below not use this?

Reply:
The goal had been to enable projection notation, `f.comm_at 5`, for which is needed to be in the right namespace. As it turns out, this wasn't working anyway. In any case, now that I'm using `abbreviation` it looks simpler.